### PR TITLE
Use auto keyword

### PR DESCRIPTION
**Pull request by suab321321**
Signed-off-by: AbhinavSingh <singhabhinav9051571833@gmail.com>

### CODE REVIEW

**Comment by suab321321** [comment]
#403 @matthew-reynolds I have made iterators auto

**Comment by matthew-reynolds** [comment]
This PR seems to do a lot of for loop stuff, which is already done in #413. Can you remove those changes, but leave your other changes to be reviewed?

**Comment by suab321321** [comment]
@matthew-reynolds I have made the `iterator` auto which is used for both `for loop` and also for `iterators` pointing to `std::find()` results.Should I make the for loop auto iterator to default ones,beacuse in some files`auto` was not used and I thought it will good to make them auto.

**Comment by bmagyar** [comment]
#413 is now merged and as a result, this PR is conflicting.

@suab321321 , for a smooth, collaborative experience, it is good to avoid modifying code that is already under review on a different PR. As @matthew-reynolds raised, you did some duplicate work that is not causing trouble on this PR. Please resolve these conflicts.

**Comment by suab321321** [comment]
@matthew-reynolds @bmagyar will update soon

**Comment by suab321321** [comment]
@matthew-reynolds @bmagyar is it okay?

File: controller_manager/src/controller_manager.cpp [hunk 1]
```diff
@@ -76,6 +76,7 @@ ControllerManager::~ControllerManager()
 void ControllerManager::update(const ros::Time& time, const ros::Duration& period, bool reset_controllers)
 {
   used_by_realtime_ = current_controllers_list_;
+  auto &controllers = controllers_lists_[used_by_realtime_];
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp]: I don't see this variable used here, can probably be removed

File: controller_manager/src/controller_manager.cpp [hunk 2]
```diff
@@ -294,15 +295,13 @@ bool ControllerManager::loadController(const std::string& name)
     try
     {
       // Trying loading the controller using all of our controller loaders. Exit once we've found the first valid loaded controller
-      std::list<ControllerLoaderInterfaceSharedPtr>::iterator it = controller_loaders_.begin();
-      while (!c && it != controller_loaders_.end())
+      for (auto const& it : controller_loaders_)
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp]: Careful here. Read the above comment carefully. I agree that this part could be improved but I am unsure at the moment how best to achieve that. Your suggested change alters the algorithm

File: controller_manager/src/controller_manager.cpp [hunk 3]
```diff
@@ -523,7 +522,7 @@ bool ControllerManager::switchController(const std::vector<std::string>& start_c
   switch_start_list_.clear();
   switch_stop_list_.clear();
 
-  std::vector<ControllerSpec> &controllers = controllers_lists_[current_controllers_list_];
+  auto &controllers = controllers_lists_[current_controllers_list_];
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp]: This could probably be const

File: controller_manager/src/controller_manager.cpp [hunk 4]
```diff
@@ -391,7 +390,7 @@ bool ControllerManager::unloadController(const std::string &name)
     }
     std::this_thread::sleep_for(std::chrono::microseconds(200));
   }
-  std::vector<ControllerSpec>
+  auto
```
Comment (bmagyar) [inline controller_manager/src/controller_manager.cpp]: I would argue that `auto` here actually hides useful information

**Review by bmagyar — CHANGES_REQUESTED** [review]

File: controller_manager/src/controller_manager.cpp (hunk 2 above)
Comment (suab321321) [reply controller_manager/src/controller_manager.cpp]:
```
for(const auto& it:controller_loaders_.begin()){
   if(c==controller_loaders_.end())
      break;
   for(const auto& cur_type:(it)->getDeclaredClass()){
      if(cur_type==type)
         c=(it)->createInstance(type);
      else
            c=it;
   }
}
```

Comment (suab321321) [reply controller_manager/src/controller_manager.cpp]: @bmagyar will this do?

Comment (suab321321) [reply controller_manager/src/controller_manager.cpp]: @bmagyar sorry I m utterly wrong did not read the code correctly.

Comment (suab321321) [reply controller_manager/src/controller_manager.cpp]: @bmagyar sir is my current implementation okay?

Comment (bmagyar) [reply controller_manager/src/controller_manager.cpp]: Let's settle for a simpler solution, auto the iterator and leave the rest as-is.

`std::list<ControllerLoaderInterfaceSharedPtr>::iterator it = controller_loaders_.begin();` -> `auto it = controller_loaders_.begin();`

**Review by bmagyar — CHANGES_REQUESTED** [review]

**Comment by suab321321** [comment]
@bmagyar

**Review by bmagyar — APPROVED** [review]
Thanks for sticking through this review!

**Pull request merged**